docs/developer-guide/index.md

```diff
@@ -10,6 +10,7 @@
    docs
    benchmarks
    continuous-integration
+   infrastructure
```

Comment:
Is this the best name,  not sure it is intuitive what it includes

Reply:
I'm not sure what other term to use, since parts of the setup (particularly on AWS) go beyond CI/CD. I didn't want to replace the CI page yet, but some of these things could go there, leaving it more focused this page more focused on the things running on AWS and GitHub Actions.

Reply:
Not sure, maybe something to indicate it is related to the CI.

Reply:
ci-cd-infrastructure?

Reply:
That would be more descriptive.